src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -292,6 +292,21 @@ std::optional<DataModel::CommandEntry> EnumeratorCommandFinder::FindCommandEntry
     return (*id == kInvalidCommandId) ? DataModel::CommandEntry::kInvalid : CommandEntryFrom(path, *id);
 }
 
+DataModel::DeviceTypeEntry DeviceTypeEntryFromEmber(const EmberAfDeviceType & other)
```

Comment:
We should consider just getting rid of EmberAfDeviceType, if it's got nothing that DeviceTypeEntry does not have.

Followup for that, though.

Reply:
Added a TODO as a followup. I would rather do this once we have only codegen data model and then we can figure out how ember can depend on the provider interfaces. Layering-wise it feels odd, since I made datamodel provider on purpose not be aware of ember. Making ember aware of datamodel provider types is not violating circularity but it may be a bit coupled.